### DISCUSSION TITLE
Create Tag as part of Pull Request Merge

### DISCUSSION THREAD

**Discussion by zlsmith86** [community/community#19629] [Pull Requests]
Tags are used to discover the build version of the code.  Many times automated builds start automatically after merge.  If a new version tag is required, it can be a race to create the tag before the build starts.  I would like to request the ability to create a tag automatically when the pull request is merged.

**Comment by lucalves** [comment]
Hi @zlsmith86, how are you? I hope you're well.

To meet your need, you can use GitHub Actions. To be more specific, you can use [GitHub Tag Action](https://github.com/marketplace/actions/github-tag). 

According to the documentation:

> A GitHub Action to automatically bump and tag master, on merge, with the latest SemVer formatted version. Works on any platform.

Could you take a look? 😉

**Reply by zlsmith86** [reply]
I am using github on prem and we do not have access to GitHub Actions

**Reply by lucalves** [reply]
> I am using github on prem and we do not have access to GitHub Actions

GitHub on prem has access to GitHub Actions. You can check how to get started via [Quickstart for GitHub Actions](https://docs.github.com/en/actions/quickstart).

In five minutes through the article you can create your first workflow using Actions.

You can verify? Anything I'm willing to help you!

**Reply by carlitos081** [reply]
I'm onprem version and I don't have GitHub Actions. @lucalves can you please show me how to do it?

From the link you posted:
![image](https://github.com/community/community/assets/5641299/30e35068-0b63-4db4-8b2c-566f283bb84a)


from my github:

![image](https://github.com/community/community/assets/5641299/3fc5b1bc-2065-41f5-8ed4-a85e1aabb3d9)
